### ISSUE TITLE
AWS CN Pricing endpoint

### ISSUE THREAD

**Issue by duweihello** [kubernetes/autoscaler#3276]
https://github.com/kubernetes/autoscaler/blob/ca6bbc96a2ef40a46043a23425bf242c5273e799/cluster-autoscaler/cloudprovider/aws/aws_util.go#L35
the pricing API is not support AWS CN regions, is there any alternative way to pass it?

**Comment by BellonXiao** [comment]
Same issue here with 1.17.2:
aws_util.go:86] Error unmarshalling https://pricing.us-east-1.amazonaws.com/offers/v1.0/aws/AmazonEC2/current/cn-north-1/index.json, skip..
aws_cloud_provider.go:363] Failed to generate AWS EC2 Instance Types: unable to load EC2 Instance Type list

**Comment by gjtempleton** [comment]
I think the ideal way to resolve this is to move away from the use of the pricing API. Alternatively you should now (as of the latest releases) be able to use the hardcoded list by using the `aws-use-static-instance-list` flag set to true as it'll no longer attempt to call the pricing API.

I'd meant to move away from the pricing APIs to resolve #2696 which includes some of the other issues with the use of the Pricing API and never got around to it earlier in the year. I'll see if I can find the time to pick this work back up.

**Comment by greenu** [comment]
related issue #3076

**Comment by momkeller** [comment]
I am also running into this issue in cn-north-1 using AS 1.15.7.  Is there a workaround or projected fix date?

**Comment by marcheil** [comment]
So what is the work around here for clusters that do not have internet access? How do we set the aws-use-static-instance-list flag?

**Comment by Jeffwan** [comment]
I approve PR https://github.com/kubernetes/autoscaler/pull/3528.
Let's wait for the next release

**Comment by jumping** [comment]
Is there any scheduler to merge the fix into some release ?

**Comment by zweizhang** [comment]
I got the same issue with the spinnaker version is 1.23.7.
And follow gjtempleton advice, I add the aws-use-static-instance-list to vaule file, and fix the CN Pricing issue.
Hoping it can help others.
```
cluster-autoscaler$ ls
Chart.yaml  README.md  README.md.gotmpl  templates  values.yaml
cluster-autoscaler$ **cat values.yaml**

**extraArgs:**
  logtostderr: true
  stderrthreshold: info
  v: 4
  # write-status-configmap: true
  # status-config-map-name: cluster-autoscaler-status
  # leader-elect: true
  # skip-nodes-with-local-storage: true
  # expander: random
  # scale-down-enabled: true
  # balance-similar-node-groups: true
  # min-replica-count: 0
  # scale-down-utilization-threshold: 0.5
  # scale-down-non-empty-candidates-count: 30
  # max-node-provision-time: 15m0s
  # scan-interval: 10s
  # scale-down-delay-after-add: 10m
  # scale-down-delay-after-delete: 0s
  # scale-down-delay-after-failure: 3m
  # scale-down-unneeded-time: 10m
  # skip-nodes-with-system-pods: true
  **aws-use-static-instance-list: true**    ------------>Add this

```

**Comment by ddelange** [comment]
fixed as of [cluster-autoscaler-1.24.0](https://github.com/kubernetes/autoscaler/releases/tag/cluster-autoscaler-1.24.0) in https://github.com/kubernetes/autoscaler/pull/4489
